python/ray/autoscaler/ray-schema.json

```diff
@@ -291,6 +291,33 @@
                        "resources": {
                            "type": "object",
                            ".*": {"type": "number"}
+                       },
+                       "initialization_commands": {
+                           "$ref": "#/definitions/commands",
+                           "description": "List of commands that will be run before `setup_commands`. If docker is enabled, these commands will run outside the container and before docker is setup."
+                       },
+                       "setup_commands": {
+                           "$ref": "#/definitions/commands",
+                           "description": "List of common shell commands to run to setup nodes."
+                       },
+                       "head_setup_commands": {
+                           "$ref": "#/definitions/commands",
+                           "description": "Commands that will be run on the head node after common setup."
+                       },
+                       "worker_setup_commands": {
+                           "$ref": "#/definitions/commands",
+                           "description": "Commands that will be run on worker nodes after common setup."
+                       },
+                       "head_start_ray_commands": {
+                           "$ref": "#/definitions/commands",
+                           "description": "Command to start ray on the head node. You shouldn't need to modify this."
+                       },
+                       "worker_start_ray_commands": {
+                          "$ref": "#/definitions/commands",
+                          "description": "Command to start ray on worker nodes. You shouldn't need to modify this."
+                       },
+                       "no_restart": {
```

Comment:
Please remove this -- I think this flag can be propagated automatically without it being part of the user schema.